src/settings/watchers/FontWatcher.ts

```diff
@@ -34,27 +37,41 @@ export class FontWatcher implements IWatcher {
     }
 
     public start() {
-        this.setRootFontSize(SettingsStore.getValue("baseFontSize"));
-        this.setSystemFont({
-            useSystemFont: SettingsStore.getValue("useSystemFont"),
-            font: SettingsStore.getValue("systemFont"),
-        });
+        this.updateFont();
         this.dispatcherRef = dis.register(this.onAction);
     }
 
     public stop() {
         dis.unregister(this.dispatcherRef);
     }
 
-    private onAction = (payload) => {
+    private updateFont() {
+        this.setRootFontSize(SettingsStore.getValue("baseFontSize"));
+        this.setSystemFont({
+            useSystemFont: SettingsStore.getValue("useSystemFont"),
+            font: SettingsStore.getValue("systemFont"),
+        });
+    }
+
+    private onAction = (payload: ActionPayload) => {
         if (payload.action === Action.UpdateFontSize) {
             this.setRootFontSize(payload.size);
         } else if (payload.action === Action.UpdateSystemFont) {
-            this.setSystemFont(payload);
+            this.setSystemFont(payload as UpdateSystemFontPayload);
+        } else if (payload.action === "on_logged_out") {
```

Comment:
Should we add an `Action.LogOut` and `Action.LogIn` ? Why is `"on_logged_out"` a plain string?

Reply:
These just haven't been brought over into the typed world yet, happy to do it as part of the PR

Reply:
Not insisting, but would be nice if possible.

Reply:
Happy to, we need to ditch the old string ones sooner or later, they make the dispatcher not type-safe